RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -371,78 +342,72 @@ reco::Muon MuonIdProducer::makeMuon( const reco::MuonTrackLinks& links )
 
 bool MuonIdProducer::isGoodTrack( const reco::Track& track )
 {
-   // Pt and absolute momentum requirement
-   if (track.pt() < minPt_ || (track.p() < minP_ && track.p() < minPCaloMuon_)){
-      LogTrace("MuonIdentification") << "Skipped low momentum track (Pt,P): " << track.pt() <<
-	", " << track.p() << " GeV";
-      return false;
-   }
+  // Pt and absolute momentum requirement
+  const double p = track.p();
+  const double pt = track.pt();
+  if (pt < minPt_ || (p < minP_ && p < minPCaloMuon_)){
+    LogTrace("MuonIdentification") << "Skipped low momentum track (Pt,P): " << pt
+                                   << ", " << track.p() << " GeV";
```

Comment:
It looks like this call to track.p() should be replaced with "p", as you did in the previous lines.